api/git/git.go

```diff
@@ -82,8 +82,17 @@ func (c gitClient) latestCommitID(ctx context.Context, opt fetchOptions) (string
 		return "", errors.Wrap(err, "failed to list repository refs")
 	}
 
+	referenceName := opt.referenceName
+	if referenceName == "" {
+		for _, ref := range refs {
+			if strings.EqualFold(ref.Name().String(), "HEAD") {
+				referenceName = ref.Target().String()
+			}
+		}
+	}
+
```

Comment:
I think we can use `"HEAD"` without coalescing it with the actual refs, as the same will be done right after. Thus this block could be updated to 
```
referenceName := opt.referenceName
if referenceName == "" {
  referenceName = "HEAD"
}
```

Reply:
The "referenceName" is expected to be the default branch name(ex. ref/heads/main) but not "HEAD" here.

Reply:
ahh...I've missed the fact that you compare `HEAD` with ref.**Name**, but then use its the ref.**Target**.